File: p3_fcfs/timer.py
import random


class Times:
    def __init__(self):
        self.estimated: int = random.randint(5, 18)
        self.elapsed: int = 0
        self.blocked: int = 0
        self.arrive: int = 0
        self.finished: int = 0
        self.response: int = 0
        self.returned: int = 0
        self.waiting: int = 0
        self.service: int = 0

    # Getters
    def get_estimated(self) -> int:
        self.estimated

    def get_elapsed(self) -> int:
        self.elapsed

    def get_blocked(self) -> int:
        self.blocked

    def get_arrive(self) -> int:
        self.arrive

    def get_finished(self) -> int:
        self.finished

    def get_response(self) -> int:
        self.response

    def get_returned(self) -> int:
        self.returned

    def get_waiting(self) -> int:
        self.waiting

    def get_service(self) -> int:
        self.service

    # Setters
    def set_estimated(self, value: int):
        self.estimated = value

    def set_elapsed(self, value: int):
        self.elapsed = value

    def set_blocked(self, value: int):
        self.blocked = value

    def set_arrive(self, value: int):
        self.arrive = value

    def set_finished(self, value: int):
        self.finished = value

    def set_response(self, value: int):
        self.response = value

    def set_returned(self, value: int):
        self.returned = value

    def set_waiting(self, value: int):
        self.waiting = value

    def set_service(self, value: int):
        self.service = value

File: p3/src/timer.rs
use rand::Rng;


#[derive(Default, Debug, Clone)]
pub struct Times { 
    estimated: u8,
    elapsed: u8,
    blocked: u8,
    arrive: u8,
    finished: u8,
    response: u8,
    returned: u8,
    waiting: u8,
    service: u8,
}

impl Times { 
    pub fn new() -> Self { 
        let mut rng = rand::thread_rng();
        Self { 
            estimated: rng.gen_range(5..=18),
            elapsed: 0,
            blocked: 0,
            arrive: 0,
            finished: 0,
            response: 0,
            returned: 0,
            waiting: 0,
            service: 0,
        }
    }

    // Getters
    pub fn get_estimated(&self) -> u8 { self.estimated }
    pub fn get_elapsed(&self) -> u8 { self.elapsed }
    pub fn get_blocked(&self) -> u8 { self.blocked }
    pub fn get_arrive(&self) -> u8 { self.arrive }
    pub fn get_finished(&self) -> u8 { self.finished }
    pub fn get_response(&self) -> u8 { self.response }
    pub fn get_returned(&self) -> u8 { self.returned }
    pub fn get_waiting(&self) -> u8 { self.waiting }
    pub fn get_service(&self) -> u8 { self.service }

    // Setters
    pub fn set_estimated(&mut self, value: u8) { self.estimated = value; }
    pub fn set_elapsed(&mut self, value: u8) { self.elapsed = value; }
    pub fn set_blocked(&mut self, value: u8) { self.blocked = value; }
    pub fn set_arrive(&mut self, value: u8) { self.arrive = value; }
    pub fn set_finished(&mut self, value: u8) { self.finished = value; }
    pub fn set_response(&mut self, value: u8) { self.response = value; }
    pub fn set_returned(&mut self, value: u8) { self.returned = value; }
    pub fn set_waiting(&mut self, value: u8) { self.waiting = value; }
    pub fn set_service(&mut self, value: u8) { self.service = value; }
}


